### DISCUSSION TITLE
Extract Sparsity to Matlab array

### DISCUSSION THREAD

**Discussion by fawidmer** [casadi/casadi#3124] [Q&A] [answered]
Hello everyone

Sorry if this question has an obvious answer, but I cannot seem to find it. I'm using CasADi in Matlab.
I have an object of the `Sparsity` class and I want to extract its contents into a native Matlab array (which is one where there are non-zero elements and zero otherwise).

Assuming the `Sparsity` object is called `spars`, I have the following ugly workaround:
```matlab
sparsStr = evalc('spars.spy()');
sparsStr = strsplit(sparsStr , '\n');
sparsStr = vertcat(sparsStr{:});
sparsityPattern = zeros(size(sparsStr));
sparsityPattern(sparsStr == '*') = 1;
```

I'm sure there is a better way than that? :)

I know about the `spy_matlab` function, but I want to extract the information programmatically.

Thanks for your help, cheers! 

Fabio

**Comment by jgillis** [comment, marked as answer]
`sparse(DM(spars,1)` should work, maybe even `sparse(spars)`. The return value is a sparse matlab matrix

**Reply by fawidmer** [reply]
Thank you! :)
